annotations.py

```diff
@@ -184,17 +189,29 @@ def to_HTML(self, header=''):
 #                         self._log_location("agn='%s'" % agn)
                         note += '<p class="note" style="{0}">{1}</p>'.format(note_style, agn)
 
-                try:
+                if agroup.highlightcolor and html_color_pattern.match(agroup.highlightcolor):
+                   msg = "Found valid HTML color '%s', using" % agroup.highlightcolor
+                   dt_bgcolor = agroup.highlightcolor
+
+                   # Currently annotations only store the foreground color. If it's not a known color, then
+                   # we should pick the background color as black or white based on the overall lightness of
+                   # the color.
+                   color = QColor(dt_bgcolor)
+                   if color.lightness() >= 128:
+                       dt_fgcolor = "#000000"
+                   else:
+                       dt_fgcolor = "#FFFFFF"
+                   self._log_location(f"HTML color lightness is {color.lightness()}, using {dt_bgcolor}")
+                elif agroup.highlightcolor in COLOR_MAP:
+                    msg = "Found match for %s in known colors" % agroup.highlightcolor
                     dt_bgcolor = COLOR_MAP[agroup.highlightcolor]['bg']
                     dt_fgcolor = COLOR_MAP[agroup.highlightcolor]['fg']
-                except:
-                    if agroup.highlightcolor is None:
-                        msg = "No highlight color specified, using Default"
-                    else:
-                        msg = "Unknown color '%s' specified" % agroup.highlightcolor
```

Comment:
If the case with `if agroup.highlightcolor is None:` is still possible, then we should keep the original logging code (the one being highlighted here) instead of new `msg = "Unknown color '%s' specified, using default" % agroup.highlightcolor` code.

Reply:
Sure I added the specific None case back